src/common/git/husky.ts

```diff
@@ -1,23 +1,64 @@
 import * as path from 'path'
 import type {NodeProject} from 'projen/lib/javascript'
 import {AssetFile} from '../files/AssetFile'
-import type {WithGitHooks} from './with-git-hooks'
+import {CheckCargoOptions, HuskyRule, WithGitHooks} from './with-git-hooks'
 
 export const addHusky = (project: NodeProject, options: WithGitHooks): void => {
   project.addDevDeps('husky')
   project.addTask('prepare', {exec: 'husky install'})
+  const huskyRules = options.huskyRules ?? [HuskyRule.COMMIT_MESSAGE]
 
-  if (options.hasDefaultCommitHook === false) {
-    return
+  huskyRules.forEach((ruleId) => {
+    switch (ruleId) {
```

Comment:
As we discussed I've retained the switch construct and added a simple condition to avoid rule rendering when a false input is provided (covered it with a test). Finally I think that using just a collection of if-statements might be even better...